test/views/declaration/ExporterEoriNumberViewSpec.scala

```diff
@@ -0,0 +1,141 @@
+
+  private val page: exporter_eori_number = instanceOf[exporter_eori_number]
+
+  private def createView(mode: Mode = Mode.Normal, form: Form[ExporterEoriNumber] = ExporterEoriNumber.form())(
+    implicit request: JourneyRequest[_]
+  ): Document =
+    page(mode, form)(request, messages)
+
+  "ExporterEoriNumber Eori Number View" should {
+    onEveryDeclarationJourney() { implicit request =>
+      val view = createView()
+      "display answer input" in {
+        val exporterEoriNumber = ExporterEoriNumber.form().fill(ExporterEoriNumber(Some(Eori("GB123456789")), YesNoAnswers.yes))
+        val view = createView(form = exporterEoriNumber)
+
+        view
+          .getElementById("Yes")
+          .getElementsByAttribute("checked")
+          .attr("value") mustBe YesNoAnswers.yes
+      }
+
+      "have proper messages for labels" in {
+        messages must haveTranslationFor("declaration.exporterEori.title")
+        messages must haveTranslationFor("declaration.exporterEori.eori.label")
+        messages must haveTranslationFor("declaration.exporterEori.hasEori.empty")
+        messages must haveTranslationFor("declaration.exporterEori.help-item1")
+      }
+
+      "display page title" in {
+        view.getElementsByClass(Styles.gdsPageLegend) must containMessageForElements("declaration.exporterEori.title")
+      }
+
+      "display section header" in {
+        view.getElementById("section-header") must containMessage("declaration.section.2")
+      }
+
+      "display eori question" in {
+        view.getElementsByClass("govuk-label--m") must containMessageForElements("declaration.exporterEori.eori.label")
+        view.getElementById("eori").attr("value") mustBe empty
+      }
+
+      "display 'Back' button that links to the correct page" in {
+        request.declarationType match {
+          case DeclarationType.CLEARANCE => {
+            val cachedParties = Parties(
+              isEntryIntoDeclarantsRecords = Some(YesNoAnswer("Yes")),
+              consignorDetails =
+                Some(ConsignorDetails(EntityDetails(None, address = Some(Address("fullName", "addressLine", "townOrCity", "postCode", "country")))))
+            )
+            val req = journeyRequest(request.cacheModel.copy(parties = cachedParties))
+
+            val view = createView()(req)
+            val backButton = view.getElementById("back-link")
+
+            backButton must containMessage("site.back")
+            backButton.getElementById("back-link") must haveHref(
+              controllers.declaration.routes.PersonPresentingGoodsDetailsController.displayPage(Mode.Normal)
+            )
+          }
+          case _ => {
+            val backButton = view.getElementById("back-link")
+
+            backButton must containMessage("site.back")
+            backButton.getElementById("back-link") must haveHref(controllers.declaration.routes.DeclarantExporterController.displayPage(Mode.Normal))
+          }
```

Comment:
I have to say that I don't like it :/
In a couple other places we have similar situation, but it is written in a way that there is a separate test case for Clearance journey. Please have a look at an [example](https://github.com/hmrc/customs-declare-exports-frontend/blob/477ea080073c6505b6b7cf19d8552c5e02773c85/test/views/declaration/DepartureTransportViewSpec.scala#L197).

Reply:
But then I need to take the `200`, `400` out of there as well and I will be left with duplicate code, unless I create a private method shared between them

Reply:
I see where you are going. Let's have a conversation about this :)

Reply:
This has been addressed. Please review.